aisdc/attacks/attack_report_formatter.py

```diff
@@ -86,51 +128,86 @@ def __init__(self, report: dict):
         self.scores = []
         self.reasons = []
 
-    def _is_svm(self, svm_weighing_score):
-        if "model" in self.report:
-            if self.report["model"] == "SVC":
-                self.scores.append(svm_weighing_score)
+    def _is_instance_based_model(self, instance_based_model_score):
+        if "model_name" in self.report:
+            if self.report["model_name"] == "SVC":
+                self.scores.append(instance_based_model_score)
                 self.reasons.append("Model is SVM (supersedes all other tests)")
+                self.immediate_rejection.append("Model is SVM")
+                return True
+            if self.report["model_name"] == "KNeighborsClassifier":
+                self.scores.append(instance_based_model_score)
+                self.reasons.append("Model is kNN (supersedes all other tests)")
+                self.immediate_rejection.append("Model is kNN")
                 return True
         return False
 
-    def _rf_min_samples_leaf(self, min_samples_leaf_score):
+    def _tree_min_samples_leaf(self, min_samples_leaf_score):
         if "model_params" in self.report:
             if "min_samples_leaf" in self.report["model_params"]:
                 min_samples_leaf = self.report["model_params"]["min_samples_leaf"]
                 if min_samples_leaf < 5:
```

Comment:
line 149: 'if min_samples_leaf <5:" should be referring to the TRE risk appetite which you read earlier from file and stored in min_samples_leaf_score.   
(maybe you want to name that variable something else? like min_samples_leaf_appetite
so the line should be:    if min_samples_leaaf < min_samples_leaf_appetite: